runtime/reconcilers/model.go

```diff
@@ -1521,3 +1515,31 @@ func md5Hash(val []byte) (string, error) {
 	}
 	return hex.EncodeToString(hash.Sum(nil)), nil
 }
+
+// shouldTriggerReset determines if a model should trigger a reset based on its change mode and the current state compared to the specified hash.
+func (r *ModelReconciler) shouldTriggerReset(model *runtimev1.Model, specHash string, exists bool) bool {
+	// These conditions trigger a reset regardless of change mode
+	if model.Spec.TriggerFull ||
+		model.State.ResultConnector == "" ||
+		model.State.RefreshedOn == nil ||
+		!exists {
+		return true
+	}
+
+	// Handle different change modes
+	switch model.Spec.ChangeMode {
+	case "manual":
+		// In manual mode, spec hash changes don't trigger automatic resets
+		// Only explicit trigger or missing output will cause a reset
+		return false
```

Comment:
1. It should return an error informing the user that the model is stopped and that they should manually trigger it. Something like `execution paused because the model definition was changed and change_mode=manual: you must manually trigger either a full or incremental refresh` (feel free to suggest something better).
2. When the user next sends a manual trigger (i.e. sets `Spec.Trigger` or `Spec.TriggerFull`), it should stop blocking reconciliation.